packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -298,17 +298,39 @@ export class TableLayout<T> extends ListLayout<T> {
       case 'rowgroup': {
         let firstVisibleRow = this.binarySearch(node.children, rect.topLeft, 'y');
         let lastVisibleRow = this.binarySearch(node.children, rect.bottomRight, 'y');
+        // Check to see if a persisted key exists before the visible rows
+        // This is for keeping focus on a row that scrolls out of view
+        for (let h = 0; h < firstVisibleRow; h++) {
+          if (this.virtualizer.isPersistedKey(node.children[h])) {
+            res.push(node.children[h].layoutInfo);
+            this.addVisibleLayoutInfos(res, node.children[h], rect);
+          }
+        }
         for (let i = firstVisibleRow; i <= lastVisibleRow; i++) {
           res.push(node.children[i].layoutInfo);
           this.addVisibleLayoutInfos(res, node.children[i], rect);
         }
+        // Check to see if a persisted key exists after the visible rows
+        for (let j = lastVisibleRow; j < node.children.length; j++) {
```

Comment:
Not entirely sure this is a problem in practice, but this for loop overlaps with the one before it (aka they both check `lastVisibleRow`). This means `res` could potentially contain the same layout info twice if the `lastVisibleRow` is a persisted key right?